examples/ui/scaling.rs

```diff
@@ -0,0 +1,142 @@
+            material: materials.add(Color::ANTIQUE_WHITE.into()),
+            ..Default::default()
+        })
+        .with_children(|parent| {
+            parent
+                .spawn_bundle(NodeBundle {
+                    style: Style {
+                        size: Size::new(Val::Px(40.), Val::Px(40.)),
+                        ..Default::default()
+                    },
+                    material: materials.add(Color::RED.into()),
+                    ..Default::default()
+                })
+                .with_children(|parent| {
+                    parent.spawn_bundle(TextBundle {
+                        text: Text::with_section("Size!", text_style, TextAlignment::default()),
+                        ..Default::default()
+                    });
+                });
+            parent.spawn_bundle(NodeBundle {
+                style: Style {
+                    size: Size::new(Val::Percent(15.), Val::Percent(15.)),
+                    ..Default::default()
+                },
+                material: materials.add(Color::BLUE.into()),
+                ..Default::default()
+            });
+            parent.spawn_bundle(ImageBundle {
+                style: Style {
+                    size: Size::new(Val::Px(30.0), Val::Px(30.0)),
+                    ..Default::default()
+                },
+                material: materials.add(asset_server.load("branding/icon.png").into()),
+                ..Default::default()
+            });
+        });
+}
+
+fn change_scaling(input: Res<Input<KeyCode>>, mut ui_scale: ResMut<TargetScale>) {
+    if input.just_pressed(KeyCode::Up) {
+        let scale = (ui_scale.target_scale * 2.0).min(8.);
+        ui_scale.set_scale(scale);
+        info!("Scaling up! Scale: {}", ui_scale.target_scale);
+    }
+    if input.just_pressed(KeyCode::Down) {
+        let scale = (ui_scale.target_scale / 2.0).max(1. / 8.);
+        ui_scale.set_scale(scale);
+        info!("Scaling down! Scale: {}", ui_scale.target_scale);
+    }
+}
+
+struct TargetScale {
+    target_scale: f64,
+    start_scale: f64,
+    target_time: Timer,
+}
+
+impl TargetScale {
+    fn set_scale(&mut self, scale: f64) {
+        self.start_scale = self.current_scale();
+        self.target_scale = scale;
+        self.target_time.reset();
+    }
+
+    fn current_scale(&self) -> f64 {
+        let completion = self.target_time.percent();
+        let multiplier = ease_in_expo(completion as f64);
+        self.start_scale + (self.target_scale - self.start_scale) * multiplier
+    }
+}
+
+fn apply_scaling(
+    time: Res<Time>,
+    mut target_scale: ResMut<TargetScale>,
+    mut ui_scale: ResMut<UiScale>,
+) {
+    let timer = target_scale.target_time.tick(time.delta());
+    if timer.finished() && !timer.just_finished() {
+        return;
+    }
+
+    ui_scale.scale = target_scale.current_scale();
+}
+
+fn ease_in_expo(x: f64) -> f64 {
+    if x == 0. {
+        0.
+    } else {
+        (2.0f64).powf(10. * x - 10.)
```

Comment:
This seems like it's too harsh - I tried `5. * x - 5.`, which was much more pleasing

Also, this scheme is non-trivial to understand without explanation (we really could do with an interpolations crate of some sort). 

Basically it's just exponential in x up to `x=1.->1`, but it took looking what it turned out like on [desmos](https://www.desmos.com/calculator/wdib1bgafd) to understand it:
![desmos-graph](https://user-images.githubusercontent.com/36049421/132961927-f09dd8e8-1813-48ab-a164-bcba8deb988a.png)

Reply:
Yeah it's just an exponential ease